tests/robustness/snapmeta/simple_test.go

```diff
@@ -9,10 +10,32 @@ func TestSimpleWithIndex(t *testing.T) {
 
 	storeKey := "key-to-store"
 	data := []byte("some stored data")
-	simple.Store(storeKey, data)
+	simple.Store(storeKey, data, nil)
+
+	checkStoredData, err := simple.Load(storeKey)
+	if err != nil {
+		t.Fatalf("expected %v keys but got %v", nil, err)
+	}
+
+	if !reflect.DeepEqual(checkStoredData, data) {
+		t.Fatalf("expected datas %v but got %v", data, checkStoredData)
+	}
 
 	idxName := "index-name"
-	simple.AddToIndex(storeKey, idxName)
+	indexUpdates := map[string]IndexOperation{
+		idxName: AddToIndexOperation,
+	}
+
+	simple.Store(storeKey, nil, indexUpdates)
+
+	checkStoredData, err = simple.Load(storeKey)
+	if err != nil {
+		t.Fatalf("expected error %v but got %v", nil, err)
+	}
+
+	if !reflect.DeepEqual(checkStoredData, []byte{}) {
+		t.Fatalf("expected data %v but got %v", data, checkStoredData)
+	}
```

Comment:
This is kinda the opposite of what I was describing, but fine. Can we at least add the case where we ensure that a `Store` with non-nil data and a non-nil index operation stores the data correctly (i.e. Load returns the data as expected)? 
And while we're at it, maybe we can add test cases for other index operations (`RemoveFromIndexOperation`, multiple operations per `Store`, operations on different indexes, concurrent store accesses) ?

Reply:
This might lend itself to a table driven test

Reply:
I have tried to initialize table-driven test, but they don't look very graceful. PTAL.
Btw, what if we merge this, run a few iterations of robustness-tests in the pipeline, and then add more tests afterward.